feature/s3/transfermanager/api_op_DownloadDirectory.go

```diff
@@ -0,0 +1,252 @@
+package transfermanager
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"strings"
+	"sync"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/s3"
+	s3types "github.com/aws/aws-sdk-go-v2/service/s3/types"
+)
+
+// DownloadDirectoryInput represents a request to the DownloadDirectory() call
+type DownloadDirectoryInput struct {
+	// Bucket where objects are downloaded from
+	Bucket string
+
+	// The destination directory to download
+	Destination string
+
+	// The S3 key prefix to use for listing objects. If not provided,
+	// all objects under a bucket will be retrieved
+	KeyPrefix string
+
+	// The s3 delimiter used to convert keyname to local filepath if it
+	// is different from local file separator
+	S3Delimiter string
+
+	// A callback func to allow users to fileter out unwanted objects
+	// according to bool returned from the function
+	Filter ObjectFilter
+
+	// A callback function to allow customers to update individual
+	// GetObjectInput that the S3 Transfer Manager generates
+	Callback GetRequestCallback
```

Comment:
You seem to be missing a few inputs from the SEP
* recursive
* failurePolicy

Reply:
Talked offline, design for these seems to be somewhat in-flux so we can defer for now.